### ISSUE TITLE
[Bug]: mo connection failed ,background log report error  txn write conflict

### ISSUE THREAD

**Issue by heni02** [matrixorigin/matrixone#6971]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.7.0 or 8b23a93):1a64c7e30d4409b5fa5ff44dc90d4ef02699b119
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior
after running bvt for one round, you can't  connect mo ,the second bvt test is as follows：
![image](https://user-images.githubusercontent.com/113406637/204272665-b921a6b0-0f0b-405a-b40b-6332a4a3afe1.png)
![image](https://user-images.githubusercontent.com/113406637/204226465-a9bbb63d-6bca-4460-b815-288318c11f2a.png)

part of log：
![image](https://user-images.githubusercontent.com/113406637/204227574-e4477a34-dfd7-4457-8f44-9227ab3e80d5.png)

**Comment by daviszhen** [comment]
The reason of "Access denied for user" is "txn write conflict".

![image](https://user-images.githubusercontent.com/60595215/204229097-19bc6421-25c7-47ae-85f3-7c68c09269f7.png)

the log shows :
```

2022/11/28 16:05:15.393612 +0800 ERROR frontend/session.go:1445 GetTxn. error:txn write conflict c0784ae6-74e2-4968-87e7-c320cbd83de9 e6dca083-bb7a-4553-82e4-61dc3c498556
github.com/matrixorigin/matrixone/pkg/frontend.(*TxnHandler).GetTxn
	/Users/heni/test-envir/matrixone/pkg/frontend/session.go:1445
github.com/matrixorigin/matrixone/pkg/frontend.(*TxnCompilerContext).getRelation
	/Users/heni/test-envir/matrixone/pkg/frontend/session.go:1561
github.com/matrixorigin/matrixone/pkg/frontend.(*TxnCompilerContext).Resolve
	/Users/heni/test-envir/matrixone/pkg/frontend/session.go:1602
github.com/matrixorigin/matrixone/pkg/sql/plan.(*QueryBuilder).buildTable
	/Users/heni/test-envir/matrixone/pkg/sql/plan/query_builder.go:1714
github.com/matrixorigin/matrixone/pkg/sql/plan.(*QueryBuilder).buildTable
	/Users/heni/test-envir/matrixone/pkg/sql/plan/query_builder.go:1813
github.com/matrixorigin/matrixone/pkg/sql/plan.(*QueryBuilder).buildTable
	/Users/heni/test-envir/matrixone/pkg/sql/plan/query_builder.go:1797
github.com/matrixorigin/matrixone/pkg/sql/plan.(*QueryBuilder).buildFrom
	/Users/heni/test-envir/matrixone/pkg/sql/plan/query_builder.go:1601
github.com/matrixorigin/matrixone/pkg/sql/plan.(*QueryBuilder).buildSelect
	/Users/heni/test-envir/matrixone/pkg/sql/plan/query_builder.go:1108
github.com/matrixorigin/matrixone/pkg/sql/plan.runBuildSelectByBinder
	/Users/heni/test-envir/matrixone/pkg/sql/plan/build.go:28
github.com/matrixorigin/matrixone/pkg/sql/plan.BuildPlan
	/Users/heni/test-envir/matrixone/pkg/sql/plan/build.go:56
github.com/matrixorigin/matrixone/pkg/sql/plan.(*BaseOptimizer).Optimize
	/Users/heni/test-envir/matrixone/pkg/sql/plan/optimize.go:45
github.com/matrixorigin/matrixone/pkg/frontend.buildPlan
	/Users/heni/test-envir/matrixone/pkg/frontend/mysql_cmd_executor.go:2356
github.com/matrixorigin/matrixone/pkg/frontend.(*TxnComputationWrapper).Compile
	/Users/heni/test-envir/matrixone/pkg/frontend/mysql_cmd_executor.go:2179
github.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).doComQuery
	/Users/heni/test-envir/matrixone/pkg/frontend/mysql_cmd_executor.go:3334
github.com/matrixorigin/matrixone/pkg/frontend.(*BackgroundHandler).Exec
	/Users/heni/test-envir/matrixone/pkg/frontend/session.go:1958
github.com/matrixorigin/matrixone/pkg/frontend.executeSQLInBackgroundSession
	/Users/heni/test-envir/matrixone/pkg/frontend/session.go:1911
github.com/matrixorigin/matrixone/pkg/frontend.(*Session).AuthenticateUser
	/Users/heni/test-envir/matrixone/pkg/frontend/session.go:1094
github.com/matrixorigin/matrixone/pkg/frontend.(*MysqlProtocolImpl).authenticateUser
	/Users/heni/test-envir/matrixone/pkg/frontend/mysql_protocol.go:1078
github.com/matrixorigin/matrixone/pkg/frontend.(*MysqlProtocolImpl).handleHandshake
	/Users/heni/test-envir/matrixone/pkg/frontend/mysql_protocol.go:1177
github.com/matrixorigin/matrixone/pkg/frontend.(*RoutineManager).Handler
	/Users/heni/test-envir/matrixone/pkg/frontend/routine_manager.go:257
github.com/fagongzi/goetty/v2.(*server).doConnection
	/Users/heni/go/pkg/mod/github.com/fagongzi/goetty/v2@v2.0.3-0.20221121133015-81355a910e4a/application.go:381
github.com/fagongzi/goetty/v2.(*server).doStart.func2.1
	/Users/heni/go/pkg/mod/github.com/fagongzi/goetty/v2@v2.0.3-0.20221121133015-81355a910e4a/application.go:344
2022/11/28 16:05:15.394063 +0800 ERROR plan/query_builder.go:1716 error: SQL parser error: table "mo_account" does not exist
github.com/matrixorigin/matrixone/pkg/sql/plan.(*QueryBuilder).buildTable
	/Users/heni/test-envir/matrixone/pkg/sql/plan/query_builder.go:1716
github.com/matrixorigin/matrixone/pkg/sql/plan.(*QueryBuilder).buildTable
	/Users/heni/test-envir/matrixone/pkg/sql/plan/query_builder.go:1813
github.com/matrixorigin/matrixone/pkg/sql/plan.(*QueryBuilder).buildTable
	/Users/heni/test-envir/matrixone/pkg/sql/plan/query_builder.go:1797
github.com/matrixorigin/matrixone/pkg/sql/plan.(*QueryBuilder).buildFrom
	/Users/heni/test-envir/matrixone/pkg/sql/plan/query_builder.go:1601
github.com/matrixorigin/matrixone/pkg/sql/plan.(*QueryBuilder).buildSelect
	/Users/heni/test-envir/matrixone/pkg/sql/plan/query_builder.go:1108
github.com/matrixorigin/matrixone/pkg/sql/plan.runBuildSelectByBinder
	/Users/heni/test-envir/matrixone/pkg/sql/plan/build.go:28
github.com/matrixorigin/matrixone/pkg/sql/plan.BuildPlan
	/Users/heni/test-envir/matrixone/pkg/sql/plan/build.go:56
github.com/matrixorigin/matrixone/pkg/sql/plan.(*BaseOptimizer).Optimize
	/Users/heni/test-envir/matrixone/pkg/sql/plan/optimize.go:45
github.com/matrixorigin/matrixone/pkg/frontend.buildPlan
	/Users/heni/test-envir/matrixone/pkg/frontend/mysql_cmd_executor.go:2356
github.com/matrixorigin/matrixone/pkg/frontend.(*TxnComputationWrapper).Compile
	/Users/heni/test-envir/matrixone/pkg/frontend/mysql_cmd_executor.go:2179
github.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).doComQuery
	/Users/heni/test-envir/matrixone/pkg/frontend/mysql_cmd_executor.go:3334
github.com/matrixorigin/matrixone/pkg/frontend.(*BackgroundHandler).Exec
	/Users/heni/test-envir/matrixone/pkg/frontend/session.go:1958
github.com/matrixorigin/matrixone/pkg/frontend.executeSQLInBackgroundSession
	/Users/heni/test-envir/matrixone/pkg/frontend/session.go:1911
github.com/matrixorigin/matrixone/pkg/frontend.(*Session).AuthenticateUser
	/Users/heni/test-envir/matrixone/pkg/frontend/session.go:1094
github.com/matrixorigin/matrixone/pkg/frontend.(*MysqlProtocolImpl).authenticateUser
	/Users/heni/test-envir/matrixone/pkg/frontend/mysql_protocol.go:1078
github.com/matrixorigin/matrixone/pkg/frontend.(*MysqlProtocolImpl).handleHandshake
	/Users/heni/test-envir/matrixone/pkg/frontend/mysql_protocol.go:1177
github.com/matrixorigin/matrixone/pkg/frontend.(*RoutineManager).Handler
	/Users/heni/test-envir/matrixone/pkg/frontend/routine_manager.go:257
github.com/fagongzi/goetty/v2.(*server).doConnection
	/Users/heni/go/pkg/mod/github.com/fagongzi/goetty/v2@v2.0.3-0.20221121133015-81355a910e4a/application.go:381
github.com/fagongzi/goetty/v2.(*server).doStart.func2.1
	/Users/heni/go/pkg/mod/github.com/fagongzi/goetty/v2@v2.0.3-0.20221121133015-81355a910e4a/application.go:344
2022/11/28 16:05:15.394231 +0800 ERROR frontend/util.go:509 SQL parser error: table "mo_account" does not exist {"session_info": ""}
```

**Comment by heni02** [comment]
run again，log ：
[log2.zip](https://github.com/matrixorigin/matrixone/files/10103624/log2.zip)

**Comment by fengttt** [comment]
run again, means?   you run again and reproed?   run from a clean install or from existing failed cluster?   actually, why do you run again?

**Comment by heni02** [comment]
sorry，don't explain clearly，xupeng has doubts about the log and needs to reproduce it again. This is the log after cleaning the directory, recompiling and reproducing

**Comment by aptend** [comment]
I repro this problem with debug level. this log is truncated and prints the conflicted entries and timestamps.

[log.zip](https://github.com/matrixorigin/matrixone/files/10111199/log.zip)

I add some comments (lines starting with ///) to present the path leading to the error.

the root cause is that dn may send duplicated history versions in `mo-database` checkpoint and cn doesn't handle it correctly. I think it is ok for cn to ignore this error and the historical versions. I will try to fix it

**Comment by aptend** [comment]
fixed in 3e3dda573d8216282df061dbce237c49c28bdfa7. @heni02 pls check

**Comment by heni02** [comment]
The passing rate of bvt case in two runs is 100%
confirm,closed
commit id: 2c5f33aced405da69c95db1a44f53382a58b96e1
